### PR TITLE
k8s: add cronjob api

### PR DESCRIPTION
**Pull request by rithujohn191**
### Description

This PR adds APIs to Describe and Delete a K8s cronjob object.

### Testing Performed
- Unit tests 
- Manual tests with a local k8s cluster:
```
$ kubectl get cronjobs -n envoy-staging
NAME           SCHEDULE      SUSPEND   ACTIVE   LAST SCHEDULE   AGE
cron-staging   */1 * * * *   False     0        <none>          19s

$ curl -X POST   http://localhost:3000/v1/k8s/describeCronJob    -H 'Content-Type: application/json'   -d '{
> "clientset": "kind-clutch-local",
> "cluster": "kind-clutch-local",
> "namespace": "envoy-staging",
> "name": "cron-staging"
> }'
{"cronjob":{"cluster":"kind-clutch-local","namespace":"envoy-staging","name":"cron-staging","schedule":"","labels":{},"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"batch/v1beta1\",\"kind\":\"CronJob\",\"metadata\":{\"annotations\":{},\"name\":\"cron-staging\",\"namespace\":\"envoy-staging\"},\"spec\":{\"jobTemplate\":{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"args\":[\"/bin/sh\",\"-c\",\"date; echo Hello from the Kubernetes cluster\"],\"image\":\"busybox\",\"imagePullPolicy\":\"IfNotPresent\",\"name\":\"hello\"}],\"restartPolicy\":\"OnFailure\"}}}},\"schedule\":\"*/1 * * * *\"}}\n"}}}

$ curl -X POST   http://localhost:3000/v1/k8s/deleteCronJob    -H 'Content-Type: application/json'   -d '{
"clientset": "kind-clutch-local",
"cluster": "kind-clutch-local",
"namespace": "envoy-staging",
"name": "cron-staging"
}'
{}

$ kubectl get cronjobs -n envoy-staging
No resources found.



```

### CODE REVIEW

File: backend/service/k8s/cronjob.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	opts := metav1.DeleteOptions{}
+	return cs.BatchV1beta1().CronJobs(cs.Namespace()).Delete(ctx, name, opts)
+}
+
+func ProtoForCronJob(cluster string, k8scronJob *v1beta1.CronJob) *k8sapiv1.CronJob {
+	clusterName := k8scronJob.ClusterName
+	if clusterName == "" {
+		clusterName = cluster
+	}
+	return &k8sapiv1.CronJob{
```
Comment (mcutalo88) [inline backend/service/k8s/cronjob.go:49]: Missing `Schedule` here

File: backend/service/k8s/cronjob_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+			Annotations: map[string]string{"test": "bar"},
+		},
+	}
+
+	return fake.NewSimpleClientset(cron)
+}
+
+func TestDescribeCron(t *testing.T) {
+	cs := testCronClientset()
+	s := &svc{
```
Comment (mcutalo88) [inline backend/service/k8s/cronjob_test.go:29]: Reduce code duplication by refactoring `testCronClientset` to return the service, also renaming the function to reflect the new behavior.

File: backend/service/k8s/cronjob.go (hunk 1 above)
Comment (rithujohn191) [reply backend/service/k8s/cronjob.go:49]: Thanks for catching that. Fixed :)

File: backend/service/k8s/cronjob_test.go (hunk 2 above)
Comment (rithujohn191) [reply backend/service/k8s/cronjob_test.go]: Refactored :)

**Review by mcutalo88 — APPROVED** [review]

**Pull request merged**